core/src/test/scala/unit/kafka/controller/ControllerIntegrationTest.scala

```diff
@@ -783,14 +788,13 @@ class ControllerIntegrationTest extends ZooKeeperTestHarness {
       TestUtils.waitUntilTrue(() => !controller.isActive, "Controller fails to resign")
 
       // Expect to capture the ControllerMovedException in the log of ControllerEventThread
-      val event = appender.getMessages.find(e => e.getLevel == Level.INFO
-        && e.getThrowableInformation != null
-        && e.getThrowableInformation.getThrowable.getClass.getName.equals(classOf[ControllerMovedException].getName))
+      logCaptureContext.await(30, TimeUnit.SECONDS)
```

Comment:
It's a shame about the new asynchrony here, but I don't see an obvious we of avoiding it.